frontend/src/old-pages/Clusters/Instances.tsx

```diff
@@ -42,77 +42,94 @@ import {InstanceStatusIndicator} from '../../components/Status'
 import EmptyState from '../../components/EmptyState'
 import DateView from '../../components/date/DateView'
 
-function InstanceActions({
-  fleetStatus,
-  instance,
-}: {
-  fleetStatus: ComputeFleetStatus
-  instance: Instance
-}) {
-  const pending = useState(['app', 'clusters', 'action', 'pending'])
-  const clusterName = useState(['app', 'clusters', 'selected'])
-  const navigate = useNavigate()
-  const logHref = `/clusters/${clusterName}/logs?instance=${instance.instanceId}`
+function InstanceActions({instance}: {instance?: Instance}) {
   const {t} = useTranslation()
+  const navigate = useNavigate()
+
+  const clusterName = useState(['app', 'clusters', 'selected'])
+  const fleetStatus: ComputeFleetStatus = useState([
+    'clusters',
+    'index',
+    clusterName,
+    'computeFleetStatus',
+  ])
+  const logHref = `/clusters/${clusterName}/logs?instance=${
+    instance && instance.instanceId
+  }`
+  const [startPending, setStartPending] = React.useState(false)
+  const [stopPending, setStopPending] = React.useState(false)
+
+  const resetPending = () => {
+    setStopPending(false)
+    setStartPending(false)
+  }
 
   const refresh = () => {
     const clusterName = getState(['app', 'clusters', 'selected'])
     clusterName &&
-      GetClusterInstances(clusterName, () =>
-        clearState(['app', 'clusters', 'action', 'pending']),
-      )
+      GetClusterInstances(clusterName, () => {
+        resetPending
+      })
   }
 
-  const stopInstance = (instance: Instance) => {
-    setState(['app', 'clusters', 'action', 'pending'], true)
-    Ec2Action([instance.instanceId], 'stop_instances', refresh)
-  }
+  const stopInstance = React.useCallback(
+    (instance: Instance) => {
+      setStopPending(true)
+      Ec2Action([instance.instanceId], 'stop_instances', refresh)
+    },
+    [instance],
+  )
 
-  const startInstance = (instance: Instance) => {
-    setState(['app', 'clusters', 'action', 'pending'], true)
-    Ec2Action([instance.instanceId], 'start_instances', refresh)
-  }
+  const startInstance = React.useCallback(
+    (instance: Instance) => {
+      setStartPending(true)
+      Ec2Action([instance.instanceId], 'start_instances', refresh)
+    },
+    [instance],
+  )
+
+  const isComputeFleetStopped = React.useMemo(
+    () => fleetStatus === ComputeFleetStatus.Stopped,
+    [fleetStatus],
+  )
+  const isHeadNodeRunning = React.useMemo(
+    () =>
+      instance &&
+      instance.nodeType === NodeType.HeadNode &&
+      instance.state === InstanceState.Running,
+    [instance],
+  )
+  const isHeadNodeStopped = React.useMemo(
+    () =>
+      instance &&
+      instance.nodeType === NodeType.HeadNode &&
+      instance.state === InstanceState.Stopped,
+    [instance],
+  )
```

Comment:
suggestion: useMemo is not needed here for two reasons:

- the value is not passed as props to a Component
- it is a boolean value, thus its reference does not change across re-renders

Reply:
Thanks, going to remove